apps/frontend/src/components/global/upload_tabs/GroupFileList.vue

```diff
@@ -0,0 +1,228 @@
+import DeleteDialog from '@/components/generic/DeleteDialog.vue';
+import ShareEvaluationButton from '@/components/generic/ShareEvaluationButton.vue';
+import EditEvaluationModal from '@/components/global/upload_tabs/EditEvaluationModal.vue';
+import {SnackbarModule} from '../../../store/snackbar';
+import {ServerModule} from '../../../store/server';
+
+@Component({
+    components: {
+        DeleteDialog,
+        EditEvaluationModal,
+        ShareEvaluationButton,
+        TagRow
+    }
+})
+export default class GroupFileList extends Vue {
+    @Prop({required: true}) readonly files!: IEvaluation[];
+    @Prop({type: Boolean, default: false}) readonly loading!: boolean;
+
+    search = '';
+    activeItem!: IEvaluation;
+    selectedFiles: IEvaluation[] = [];
+    deleteItemDialog = false;
+    convertedFiles: (IEvaluation & {group: string})[] = []
+    headers: Object[] = [
+        {
+            text: 'Filename',
+            align: 'start',
+            sortable: true,
+            value: 'filename'
+        },
+        {
+            text: 'Tags',
+            value: 'evaluationTags',
+            sortable: true
+        },
+        {
+            text: 'Group',
+            value: 'group',
+            sortable: true
+        },
+        {
+            text: 'Uploaded',
+            value: 'createdAt',
+            sortable: true
+        },
+        {
+            text: 'Actions',
+            value: 'actions',
+            align: 'right'
+        }
+    ];
+
+    @Watch('files')
+    onUpdateFiles() {
+        this.convertFilesToGroups();
+    }
+
+    async updateEvaluations() {
+        await EvaluationModule.getAllEvaluations();
+        this.convertFilesToGroups();
+    }
+
+    load_results(evaluations: IEvaluation[]) {
+        this.selectedFiles = [];
+        this.$emit('load-results', evaluations);
+    }
+
+    convertFilesToGroups() {
+        const foundGroups: IGroup[] = [];
+        this.convertedFiles = [];
+        // Filter out all unique groups
+        this.files.forEach((file) => {
+            if(file.groups?.length !== 0){
+                file.groups?.forEach((group) => {
+                    if(!foundGroups.some((foundGroup) => foundGroup.id === group.id)){
+                        foundGroups.push(group);
+                    }
+                    foundGroups.forEach((foundGroup) => {
+                        if(foundGroup.id === group.id) {
+                            this.convertedFiles.push({...file, group: group.name});
+                        }
+                    })
+                })
+            } else {
+              if(file.userId === ServerModule.userInfo.id){
+                this.convertedFiles.push({...file, group: 'My Files'});
+              } else if(file.public) {
+                this.convertedFiles.push({...file, group: 'Public Files'});
+              } else {
+                this.convertedFiles.push({...file, group: 'No Group'});
+              }
```

Comment:
It currently looks a little strange having the word `'group'` in front of My Files, Public Files, and 'No Group', is there any way to remove that for these?